RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -325,6 +326,35 @@ - (void)mapView:(MGLMapView *)mapView annotation:(id<MGLAnnotation>)annotation c
         [_eventDispatcher sendInputEventWithName:RCTMGLOnRightAnnotationTapped body:event];
     }
 }
+
+- (MGLAnnotationImage *)mapView:(MGLMapView *)mapView imageForAnnotation:(id<MGLAnnotation>)annotation
+{
+    NSString *id = [(RCTMGLAnnotation *) annotation id];
+    NSString *url = [(RCTMGLAnnotation *) annotation annotationImageURL];
+    CGSize imageSize = [(RCTMGLAnnotation *) annotation annotationImageSize];
+    MGLAnnotationImage *annotationImage = [mapView dequeueReusableAnnotationImageWithIdentifier:id];
+    
+    if (!annotationImage) {
+        
+        UIImage *image = nil;
+        if ([url hasPrefix:@"image!"]) {
+            NSString* localImagePath = [url substringFromIndex:6]; //image!myImage.jpg
+            image = [UIImage imageNamed:localImagePath];
+        } else {
+            image = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:url]]];
+        }
+
+        
+        UIGraphicsBeginImageContextWithOptions(imageSize, NO, 0.0);
+        [image drawInRect:CGRectMake(0, 0, imageSize.width, imageSize.height)];
```

Comment:
Make sure to test this where `@1x` and `@2x` images with the same name are provided and specified by name, for applications that target both Retina and non-Retina devices.